modules/internal/String.chpl

```diff
@@ -163,6 +211,23 @@ module String {
     var shortData : chpl__inPlaceBuffer;
   }
 
+  /*
+     A value of type :record:`codePointIndex` can be passed to several
+     `string` functions to indicate that the function should operate
+     with units of code points. See for example :proc:`~string.this`.
```

Comment:
The only thing we can do with a `codePointIndex` is pass it to `this()` and cast it back and forth with int.  I don't think we should say several here.